### ISSUE TITLE
no case clause matching {error,closed}

### ISSUE THREAD

**Issue by nightwolfz** [emqx/emqx#3268]
#### Environment

- OS: Debian 10
- Erlang/OTP: 
- EMQ:  v4.0.1

#### Description

Please help, I'm getting these messages every few minutes and its making my system load skyrocket.

```
2020-02-23 20:45:46.577 [error] <<"akasy_4876c348">>@192.168.100.3:36808   crasher:
    initial call: cowboy_clear:connection_process/4
    pid: <0.16511.2>
    registered_name: []
    exception error: no case clause matching {error,closed}
      in function  cowboy_websocket:websocket_send_close/2 (/emqx-rel/_build/emqx-pkg/lib/cowboy/src/cowboy_websocket.erl, line 640)
      in call from cowboy_websocket:websocket_close/3 (/emqx-rel/_build/emqx-pkg/lib/cowboy/src/cowboy_websocket.erl, line 623)
    ancestors: [<0.1742.0>,<0.1741.0>,ranch_sup,<0.1311.0>]
    message_queue_len: 16
    messages:  .........................BIG BINARY MESSAGE HERE..................
```

Location of the error:  https://github.com/ninenines/cowboy/blob/master/src/cowboy_websocket.erl#L639

What could be the issue?

**Comment by tigercl** [comment]
@nightwolfz Could you describe the scene in more detail? We can use this information to locate the issue better.

**Comment by nightwolfz** [comment]
@tigercl 

I'm broadcasting relatively big binary messages (Up to 200kb+) to a few thousand clients. On a fixed regular interval, I seem to be getting these huge spikes in system load. I'm not sure what's causing it.

![image](https://user-images.githubusercontent.com/805022/75609027-725ec900-5b05-11ea-84e1-a07e17ecee7a.png)

Is there anything I could look into? I increased max_packet_size but the issue is still there

**Comment by HJianBo** [comment]
Hi @ngjaying We have fixed and released at v4.0.2. You can try it later

**Comment by ngjaying** [comment]
@nightwolfz forward this to you as @HJianBo refer to me incorrectly

> Hi @ngjaying We have fixed and released at v4.0.2. You can try it later
